### ISSUE TITLE
Question: Reducing False Positives with Empty annotation Images in training process

### ISSUE THREAD

**Issue by xuechaoyan** [Peterande/D-FINE#277]
Question: Reducing False Positives with Empty Images

Dear friends and code authors,

Thank you for sharing this codebase and ideas!

I'm seeing a high rate of false positives with my trained object detection model.

My question is: Can adding images with no objects (empty labels) to the training data help reduce these false positives?

If so, could you briefly explain why? If not, could you explain why this approach isn't effective?

Thanks for any insight!

Best wish

**Comment by xuechaoyan** [comment]
亲爱的作者大大，我在尝试使用空标签的图像来训练的时候会在epoch1卡住，但是也不报错，是不是我们的算法不能使用空标签的图片来训练？QAQ求指导一下。

**Comment by PKurnikov** [comment]
Yes, this approach works because the loss will be computed on false activations, which encourages the model to make fewer mistakes, assuming the data is homoscedastic and from similar domains. However, to prevent hangs, the following fix is needed in the code:
https://github.com/Peterande/D-FINE/issues/247

**Comment by xuechaoyan** [comment]
> Yes, this approach works because the loss will be computed on false activations, which encourages the model to make fewer mistakes, assuming the data is homoscedastic and from similar domains. However, to prevent hangs, the following fix is needed in the code: [#247](https://github.com/Peterande/D-FINE/issues/247)

Thank you so much!!!!

**Comment by guods** [comment]
添加空白图片参与训练和denoising.py在与图片原有目标没有交集的区域随机生成目标，是可以缓解误报，但对部分数据集还是没有很好的解决误报问题。如果在dec_socre_head中与detr类似类别数量增加一类当作背景，把随机生成的框标签视为背景类，在分类时会不会把之前与待检测目标不相似的误报归类于背景，减少误报的产生，对检出率影响有多大

**Comment by guods** [comment]
[https://github.com/ArgoHA/custom_d_fine/blob/main/src/d_fine/arch/dfine_decoder.py#L559](url) 这个似乎考虑到背景类

**Comment by xuechaoyan** [comment]
> 添加空白图片参与训练和denoising.py在与图片原有目标没有交集的区域随机生成目标，是可以缓解误报，但对部分数据集还是没有很好的解决误报问题。如果在dec_socre_head中与detr类似类别数量增加一类当作背景，把随机生成的框标签视为背景类，在分类时会不会把之前与待检测目标不相似的误报归类于背景，减少误报的产生，对检出率影响有多大

我在我的实验中是并没有设定多一个背景的类，这个误报缓解具体看你的误报图片，你可以把你的结果生成出来看，看看算法具体在哪些地方，或者在什么图片情况下容易误报，然后把这类的图片当成没有标签的图片拿去训练。在我的实验场景下，这样可以降低很多的误报

**Comment by 51hhh** [comment]
> 添加空白图片参与训练和denoising.py在与图片原有目标没有交集的区域随机生成目标，是可以缓解误报，但对部分数据集还是没有很好的解决误报问题。如果在dec_socre_head中与detr类似类别数量增加一类当作背景，把随机生成的框标签视为背景类，在分类时会不会把之前与待检测目标不相似的误报归类于背景，减少误报的产生，对检出率影响有多大

我参照你的思路尝试实现


从左上，右上，左下，右下依次为：
exp_s_finetune_neg_aug exp_s_finetune_neg_objtococo_aug
exp_s_transfer_aug_2000 exp_s_transfer_objtococo_aug_2000

<img width="3042" height="1129" alt="Image" src="https://github.com/user-attachments/assets/c6ba82cf-689c-408e-b1a7-fc76fd6e3798" />

```
┌────────────────────────────┬──────────┬────────────┬───────────┬───────────┬───────────┬────────┬────────────┐
│            模型            │  预训练  │ 负样本微调 │ FPPI @0.3 │ FPPI @0.5 │ FPPI @0.7 │ Max FP │ Clean @0.3 │
├────────────────────────────┼──────────┼────────────┼───────────┼───────────┼───────────┼────────┼────────────┤
│ transfer_aug               │ obj      │ 否         │ 2.15      │ 0.83      │ 0.33      │ 0.930  │ 0%         │
├────────────────────────────┼──────────┼────────────┼───────────┼───────────┼───────────┼────────┼────────────┤
│ transfer_objtococo_aug     │ obj+coco │ 否         │ 1.11      │ 0.48      │ 0.32      │ 0.938  │ 27.7%      │
├────────────────────────────┼──────────┼────────────┼───────────┼───────────┼───────────┼────────┼────────────┤
│ finetune_neg_aug           │ obj      │ 是         │ 2.36      │ 0.68      │ 0.22      │ 0.885  │ 0%         │
├────────────────────────────┼──────────┼────────────┼───────────┼───────────┼───────────┼────────┼────────────┤
│ finetune_neg_objtococo_aug │ obj+coco │ 是         │ 0.93      │ 0.45      │ 0.18      │ 0.925  │ 36.6%      │
└────────────────────────────┴──────────┴────────────┴───────────┴───────────┴───────────┴────────┴────────────┘
``` 
                            obj365                    obj2coco 
                  ──────────          ─────────────
无负样本:          2.15 FPPI           1.11 FPPI
                     ↓ 负样本微调         ↓ 负样本微调
有负样本:          2.36 FPPI           0.93 FPPI 

**我的背景图片只添加101张，对于总数据集2000张明显过少，所有可能导致效果不明显**




上图为exp_s_obj2coco_neg_d，下图为exp_s_obj2coco_neg_d_q100

<img width="2308" height="1692" alt="Image" src="https://github.com/user-attachments/assets/bf3435b8-eec6-4c2b-9fe4-0a6ef5eed7a7" />


```
┌───────────────────┬──────────┬──────────┬────────────────┬──────────┬──────────┬────────────────┐
│       模型        │   策略   │ FPPI@0.3 │ Clean Rate@0.3 │ Max Conf │ FPPI@0.5 │ Clean Rate@0.5 │
├───────────────────┼──────────┼──────────┼────────────────┼──────────┼──────────┼────────────────┤
│ obj2custom (基准) │ 无       │ 1.109    │ 27.7%          │ 0.938    │ 0.475    │ 58.4%          │
├───────────────────┼──────────┼──────────┼────────────────┼──────────┼──────────┼────────────────┤
│ finetune_neg_aug  │ 无D      │ 0.931    │ 36.6%          │ 0.925    │ 0.446    │ 60.4%          │
├───────────────────┼──────────┼──────────┼────────────────┼──────────┼──────────┼────────────────┤
│ neg_d             │D      │ 0.040    │ 96.0%          │ 0.417    │ 0.000    │ 100%           │
├───────────────────┼──────────┼──────────┼────────────────┼──────────┼──────────┼────────────────┤
│ neg_d_q100        │D+Q100 │ 0.050    │ 96.0%          │ 0.698    │ 0.010    │ 99.0%          │
└───────────────────┴──────────┴──────────┴────────────────┴──────────┴──────────┴────────────────┘
```


目前测试neg_d 最优的，也不需要额外添加数据。
